.github/workflows/ci.yml

```
@@ -89,13 +89,35 @@ jobs:
 
-      - name: codegen (smoke)
-        # Promote to `pnpm run codegen --check` once Sprint 1 emits real output.
+      - uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: src-tauri
+
+      - name: codegen (regenerate Rust → TS bindings)
         run: pnpm run codegen
 
+      - name: codegen-diff gate (no uncommitted Rust → TS drift)
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2006:style:2:64: Use $(...) notation instead of legacy backticks `...`

(shellcheck)